dlt/pipeline/pipeline.py

```diff
@@ -638,11 +638,40 @@ def run(
         Returns:
             LoadInfo: Information on loaded data including the list of package ids and failed job statuses. Please not that `dlt` will not raise if a single job terminally fails. Such information is provided via LoadInfo.
         """
+
+        def ensure_loader_file_format(
```

Comment:
The current setup uses Parquet "staging" files that are written to a Delta table. If a user sets `table_format` to `delta`, then we set `loader_file_format` to `parquet`, overriding the value configured by the user (if any).

Reply:
please check my overall review. if we go for delta as a destination we could still keep this as a file format and this function could: (if file format is delta)
1. check if destination is `delta` and exit
2. check if destination is `filesystem` and swap it for `delta`
maybe let's write first version of the code and then we'll decide.